src/DGtal/math/linalg/IntegerMatrixFunctions.h

```diff
@@ -0,0 +1,471 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file IntegerMatrixFunctions.h
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2025/09/01
+ *
+ * Header file gathering functions specific to integer matrix
+ * computations, like specific determinant or basis reduction via LLL.
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(IntegerMatrixFunctions_RECURSES)
+#error Recursive header files inclusion detected in IntegerMatrixFunctions.h
+#else // defined(IntegerMatrixFunctions_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define IntegerMatrixFunctions_RECURSES
+
+#if !defined IntegerMatrixFunctions_h
+/** Prevents repeated inclusion of headers. */
+#define IntegerMatrixFunctions_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/kernel/ArithmeticConversionTraits.h"
+#include "DGtal/arithmetic/IntegerComputer.h"
+#include "DGtal/math/linalg/SimpleMatrix.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  namespace functions {
+
+    /// Negates the components of the input vector.
+    ///
+    /// @tparam TComponent the scalar type of each component.
+    /// @param[in,out] V the vector as input, outputed as -V.
+    template <typename TComponent>
+    void
+    negate( std::vector<TComponent> &V );
+
+    /// Returns 'true' iff all the components of the two  vectors are equal.
+    ///
+    /// @tparam TComponent the scalar type of each component.
+    ///
+    /// @param[in] a the left vector (same size)
+    /// @param[in] b the right vector (same size)
+    ///
+    /// @return 'true' iff all the components of the two  vectors are equal.
+    template <typename TComponent>
+    bool
+    equals( const std::vector<TComponent> &a, const std::vector<TComponent> &b );
```

Comment:
I understand why you do that but this may induce a confection between these functions, and methods of (static dim) PointVector (that also handles type overflow conversion) ...

Reply:
(not sure if SimpleMatrix is documented in a user doc btw...)

Reply:
Yes SimpleMatrix is not really documented in a user doc. Otherwise, this functions are different from PointVector methods (and I cannot use PointVector methods) because the size of vectors may vary.